test/modules/core/shaderlib/shadow/shadow.spec.js

```diff
@@ -57,8 +43,25 @@ test('shadow#getUniforms', t => {
     `Shadow projection center in LNG_LAT mode is correct!`
   );
   t.deepEqual(
-    uniforms[`shadow_uViewProjectionMatrices[0]`],
-    viewProjectionMatrix.clone().translate(new Vector3(viewport.center).negate()),
+    uniforms[`shadow_uViewProjectionMatrices[0]`].toArray(),
```

Comment:
A test case is meaningless without explaining why the expected output is there. This function is not returning *correct* result, just *expected* result, if the correctness of the expected value cannot be independently validated. Next time you tweak the implementation, you will have to come here and change these tests. That is to say, these tests do not contribute to you writing more robust code.

A better way to test this would be to pick a few points inside the test viewport, e.g. the center, a near corner, a far corner, both on the ground and with altitude (to test the "high building" use case). Transform those points with your projection matrix and projection center, and then check if they are indeed visible in the shadow texture. Next time you tweak the calculation of these matrices, you will have solid test cases to prove that they still work, or work better.